### ISSUE TITLE
TypeError: Cannot assign to read only property '__entity' of object '#< ...

### ISSUE THREAD

**Issue by kyllerss** [mikro-orm/mikro-orm#3945]
**Describe the bug**
I am modeling a one-to-many relationship and am getting a cryptic error. The error refers to an entity unrelated to the one I am testing. The error is:

```
TypeError: Cannot assign to read only property '__entity' of object '#<TrackFeedEntity>'
```

The entity `TrackFeedEntity` is completely unrelated to the functionality I am looking at. I traced all imports and all definitions in the classes I am testing and have not found a trace of this class. 

I am hoping the problem is something simple, but I have been unable to figure it out.

Here are the definitions in question:

```
@Entity({tableName: "original_uploads"})
export class TrackUploadEntity extends BaseUUIDCreationAuditableEntity {

    @ManyToOne({fieldName: 'user_uuid', entity: () => UserAccountEntity, eager: true, cascade: []})
    user_account!: UserAccountEntity; 
    
    @Property({type: 'string'})
    path_name: string;
    
    @Property({type: 'string'})
    file_type: TrackUploadFileType;

    @Property({type: 'string'})
    md5: string;

    // @OneToMany({entity: () => ActivityType, mappedBy: 'track_upload', orphanRemoval: true})
    @OneToMany({entity: () => ActivityType, mappedBy: activity_type => activity_type.track_upload, orphanRemoval: true})
    activity_types = new Collection<ActivityType>(this);

    constructor(path_name: string, file_type: TrackUploadFileType, md5: string, activity_types: ActivityType[]) {

        super();
        
        this.path_name = path_name;
        this.file_type = file_type;
        this.md5 = md5;

        for (let activity_type of activity_types) {
            this.activity_types.add(activity_type);
        }
    }
}
```

The *many* part of the relation:
```
@Entity({tableName: "original_upload_activity_types"})
export class ActivityType extends BaseUUIDEntity {

    @ManyToOne({fieldName: 'original_upload_uuid', entity: () => TrackUploadEntity, eager: true, cascade: []})
    track_upload!: TrackUploadEntity; 
    
    @Property({type: 'string'})
    activity_type!: string;
    
    @Property({type: 'string'})
    activity_subtype!: string;

    static build(activity_type: string, activity_subtype: string): ActivityType {
        
        let to_return = new ActivityType();
        to_return.activity_type = activity_type;
        to_return.activity_subtype = activity_subtype;

        return to_return;
    }
}

```

Database definitions:
```
CREATE TABLE public.original_uploads (
	uuid uuid NOT NULL DEFAULT gen_random_uuid(),
	user_uuid uuid NOT NULL,
	path_name varchar NOT NULL,
	file_type varchar NOT NULL,
	creation_timestamp timestamptz NOT NULL,
	md5 varchar NOT NULL DEFAULT gen_random_uuid(),
	CONSTRAINT original_uploads_pkey PRIMARY KEY (uuid)
);
CREATE UNIQUE INDEX original_uploads_md5_idx ON public.original_uploads USING btree (md5);
CREATE INDEX original_uploads_path_name_idx ON public.original_uploads USING btree (path_name);
CREATE INDEX original_uploads_user_uuid_idx ON public.original_uploads USING btree (user_uuid, uuid);

CREATE TABLE public.original_upload_activity_types (
	uuid uuid NOT NULL DEFAULT gen_random_uuid() PRIMARY KEY,
	original_upload_uuid uuid NOT NULL,
	activity_type varchar NOT NULL,
	activity_subtype varchar NULL,
	CONSTRAINT original_upload_activity_types_fk FOREIGN KEY (original_upload_uuid) REFERENCES public.original_uploads(uuid)
);

```

A couple of the supporting classes:
```
@Entity({abstract: true})
export abstract class BaseUUIDCreationAuditableEntity extends CreationAuditableEntity {

  @PrimaryKey({ type: 'uuid', defaultRaw: 'gen_random_uuid()' })
  uuid!: string;

}

@Entity({abstract: true})
export abstract class CreationAuditableEntity {

  @Property({type: 'datetime'})
  creation_timestamp: Date = new Date();
  
}

```

**Stack trace**
```
TypeError: Cannot assign to read only property '__entity' of object '#<TrackFeedEntity>'

```

**To Reproduce**
Running a fairly straightforward integration test.
```
describe("Track uploads", () => {

    it("can be created in the database", async () => {

        let entity_manager = await em.getEM();
        let random_uuid = crypto.randomUUID();
        let user_account: UserAccountEntity;
        let user_uuid: string;

        await RequestContext.createAsync(entity_manager, async () => {

            user_account = await create_test_user_account(`username-${random_uuid}`);
            user_uuid = user_account.uuid;    

            // console.log(`Created test user w/ id: ${user_uuid}`);
        });

        let pathname = "test_path_name_1";
        let file_type: TrackUploadFileType = 'brt';
        let md5 = "md5-contents-1-2-3-4";
        let track_upload: TrackUploadEntity;

        let activity_types: ActivityType[] = [ActivityType.build("activity_type_1", "activity_subtype_1"),
                                              ActivityType.build("activity_type_2", "activity_subtype_2")];
        await RequestContext.createAsync(entity_manager, async () => {

            track_upload = await track_upload_repository.create_original_upload(user_account!, pathname, file_type, md5, activity_types);
        });

        await RequestContext.createAsync(entity_manager, async () => {

            let db_track_upload = await track_upload_repository.fetch_original_upload(track_upload!.uuid);
            expect(db_track_upload).toBeDefined();
            expect(db_track_upload?.file_type).toBe(file_type);
            expect(db_track_upload?.md5).toBe(md5);    
            expect(db_track_upload?.path_name).toBe(pathname);    
            expect(db_track_upload?.user_account).toBe(user_account);
            
            // validate activity types
            expect(db_track_upload?.activity_types).toBeDefined();

            let activity_types = db_track_upload?.activity_types.toArray();
            expect(activity_types!.length).toBe(2);
            expect(activity_types!.find(item => item.activity_type === "activity_type_1" && item.activity_subtype === "activity_subtype_1")).toBeTruthy();
            expect(activity_types!.find(item => item.activity_type === "activity_type_2" && item.activity_subtype === "activity_subtype_2")).toBeTruthy();
        });
    });

})
```

**Expected behavior**
Data to be persisted and subsequently loaded.

**Additional context**
Happy to create a reproduction, but I am hoping something simple jumps out with the info I provided.

**Versions**

| Dependency | Version |
| - | - |
| typescript | 4.9.4 |
| mikro-orm | 5.6.7 |
| postgresql | 5.6.7 |

**Comment by B4nan** [comment]
That seems like the entity is being discovered for the second time, as only during discovery we set those prototype properties. What's the stack trace?

**Comment by kyllerss** [comment]
Odd. 

There was no stack trace - only the single line that I referenced. To give you the data you were asking for I explicitly added a try-catch around the line in the test code that was presumably throwing the exception, and I found that the cause was an unrelated issue to the database code. I don't know why the exception would be swallowed up and ending up with the referenced misleading indicator. I fixed the original issue, and everything is working as expected. 

Somewhat of an unsatisfying outcome, but thank you for stirring the oxygen tanks.

I think this may be more of an issue w/ `vitest`.
